site/content/3.12/index-and-search/indexing/working-with-indexes/multi-dimensional-indexes.md

```diff
@@ -7,15 +7,47 @@ description: >-
   such as time ranges, for efficient intersection of multiple range queries
 archetype: default
 ---
-The multi-dimensional index type is called **ZKD**.
-
-{{< warning >}}
-`zkd` indexes are an **experimental** feature.
-{{< /warning >}}
-
-A multi-dimensional index is setup by setting the index type to `"zkd"`.
-The `fields` attribute describes which fields are used as dimensions.
-The value of each dimension has to be a numeric (double) value.
+A multi-dimensional index maps multi-dimensional data in the form of multiple
+numeric attributes to one dimension while mostly preserving locality so that
+similar values in all of the dimensions remain close to each other in the mapping
+to a single dimension. Queries that filter by multiple value ranges at once can
+be better accelerated with such an index compared to a persistent index.
+
+You can choose between two subtypes of multi-dimensional indexes:
+
+- An `mdi` index with a `fields` property that describes which document
+  attributes to use as dimensions
+- An `mdi-prefixed` index with a `fields` property as well as a `prefixFields`
+  property to specify one or more mandatory document attributes to narrow down
+  the search space using equality checks
+
+Both subtypes require that the attributes described by `fields` have numeric
+values. You can optionally omit documents from the index that have any of
+the `fields` or `prefixFields` attributes not set or set to `null` by declaring
+the index as sparse with `sparse: true`.
+
+Multi-dimensional indexes can be created with a uniqueness constraint with
+`unique: true`. This only allows a single document with a given combination of
+attribute values, using all of the `fields` attributes and (for `mdi-prefixed`
+indexes) `prefixFields`. Documents omitted because of `sparse: true` are exempt.
+
+You can store additional attributes in multi-dimensional indexes with the
+`storedValues` property. They can be used for projections (unlike the `fields`
+attributes) so that indexes can cover more queries without having to access the
+full documents.
```

Comment:
Just as a side node, while the persistent index does not allow having an overlap between `storedValues` and `fields`, the mdi index explicitly allows this. However, `prefixFields` and `storedValues` should not overlap. A typical pattern is to have `fields = ['x', 'y']` and `storedValues= ['x', 'y']` if you want to project from `x` and `y` later on.

Reply:
Added a link to the section